src/hotspot/share/gc/g1/g1ConcurrentRefineThread.cpp

```diff
@@ -79,6 +79,10 @@ void G1ConcurrentRefineThread::run_service() {
     } else {
       _vtime_accum = 0.0;
     }
+
+    if (UsePerfData && os::is_thread_cpu_time_supported() && is_primary()) {
+      _cr->update_concurrent_refine_threads_cpu_time();
+    }
```

Comment:
There are two classes for primary thread and secondary refinement thread: `G1PrimaryConcurrentRefineThread` and `G1SecondaryConcurrentRefineThread`. It is probably cleaner to move this part inside `G1PrimaryConcurrentRefineThread` and add a virtual method in `G1ConcurrentRefineThread`. We can get rid of the `is_primary()` check as well.

```
class G1ConcurrentRefineThread {
   virtual void possibly_update_threads_cpu_time() {};
}

void G1PrimaryConcurrentRefineThread::possibly_update_threads_cpu_time() {
  if (UsePerfData && os::is_thread_cpu_time_supported()) {
    _cr->update_concurrent_refine_threads_cpu_time();
  }
}
```

Reply:
Sounds good, let me know if this change is how you envisioned it!